src/components/letter/CheckoutForm.tsx

```diff
@@ -0,0 +1,118 @@
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+import React, { useState, ReactElement } from "react";
+import { Address, LETTER_COST } from "./types";
+
+import Alert from "react-bootstrap/Alert";
+import Button from "react-bootstrap/Button";
+
+import { loadStripe } from "@stripe/stripe-js";
+import { isTestMode } from "./utils";
+
+// Make sure to call `loadStripe` outside of a component’s render to avoid
+// recreating the `Stripe` object on every render.
+const stripePk = isTestMode()
+  ? "pk_test_51GqpRpGLGlm5kFVxzwruVzMZ2Bc07pqosMzyiZd6ixInJHEq6MgFE9v1kRVJZUUhuOT3X2XdfHj31oknZEmKK6KT004CUm09hp"
+  : "pk_live_51v9NZmT3TbTBBYultfXFkXO00Ohhh09jN";
+
+const stripePromise = loadStripe(stripePk);
+
+const CheckoutForm = ({
+  checkedAddresses,
+  myAddress,
+  body,
+  formValid,
+  email,
+  variables,
+}: {
+  checkedAddresses: Address[];
+  myAddress: Address;
+  body: string;
+  formValid: boolean;
+  email: string;
+  variables: Record<string, string>;
+}): ReactElement => {
```

Comment:
I don't really see a clear way to break this down further that doesn't do a ton of prop-drilling/passing to get there which I don't think helps right now.

I moved totalAmount into the makeButtonText function, buttonText is no longer a variable, isDisabled is still there.

I'm not sure what you mean about refactoring out the initialization logic ... the useState() bits are just what new functional react components look like. And hooks really should be at the top of the function (because it's a requirement that you always have the same hooks generated in a functional component - if you don't put all that at the top, you run the risk of having an early return that means you're making fewer hooks on some runs than others)